### DISCUSSION TITLE
Implementing Linear threshold function (or PiecewiseLinear) as a function of time

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#17729] [Q&A General] [answered]
Hi guys, I wonder how can I implement Linear threshold function as a function of time in moose?

for example in (t,f) coordinate,

I want f = 1 from t = 0 to 1 
f = linear line with coordinate of (1,1) to (1.5, 1e-4) from t>1 to t = 1.5
then f = 1e-4 from t>1.5

I know I could do it with `type = ParsedFunction` with something similar to this 
`'if(t<5,(-9.81*2700*(1000-z))*(t/5),-9.81*2700*(1000-z))'`

also, I know there's PiecewiseLinear function, but I'm not sure how to implement it with t (current simulation time).
is there anything as simple as

```
 type = PiecewiseLinear
      value = '0 1 1e-4 1e-4'
          t = '0 1 1.5 inf'
```

?

Thank you.

Kind regards,
Traiwit

![linearthreshold function](https://user-images.githubusercontent.com/79562439/116849352-ee84fd00-ac31-11eb-9ff4-09006f33be6e.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

As mentioned in the documentation for the PiecewiseLinear function, the default x-data corresponds to time. So you may define your functions in one of the three ways outlined.

https://mooseframework.inl.gov/source/functions/PiecewiseLinear.html

Best,
Guillaume

**Reply by Traiwit** [reply]
got it thanks @GiudGiud 

```
[./density_excav6]
  type = PiecewiseLinear
    x = '0 5 5.5 6 6.5 11'
    y = '2700 2700 1.225 1.225 1500 1500'
[../]
[./density_excav7]
  type = PiecewiseLinear
    x = '0 7 7.5 8 8.5 11'
      y = '2700 2700 1.225 1.225 1500 1500'
[../]
[./density_excav8]
  type = PiecewiseLinear
  x = '0 9 9.5 10 10.5 11'
    y = '2700 2700 1.225 1.225 1500 1500'
[../]
```
